components/datasource-selector/micka/micka.service.ts

```diff
@@ -0,0 +1,335 @@
+   * @param {object} dataset Configuration of selected datasource (from app config)
+   * @param {object} query Container for all query filter values
+   * @param {Function} extentFeatureCreated Function which gets called
+   * @param {string} textField Name of the field to search in
+   * extent feature is created. Has one parameter: feature
+   * @description Loads datasets metadata from selected source (CSW server).
+   * Currently supports only "Micka" type of source.
+   * Use all query params (search text, bbox, params.., sorting, start)
+   */
+  queryCatalog(
+    dataset: HsEndpoint,
+    query,
+    extentFeatureCreated,
+    textField: string
+  ): void {
+    const b = transformExtent(
+      this.hsMapService.map
+        .getView()
+        .calculateExtent(this.hsMapService.map.getSize()),
+      this.hsMapService.map.getView().getProjection(),
+      'EPSG:4326'
+    );
+    const bbox = this.hsMickaFilterService.filterByExtent
+      ? "BBOX='" + b.join(' ') + "'"
+      : '';
+    const text =
+      query.textFilter !== undefined && query.textFilter.length > 0
+        ? query.textFilter
+        : query.title;
+    const sql = [
+      text != '' ? `${textField} like '*${text}*'` : '',
+      bbox,
+      //param2Query('type'),
+      this.param2Query('ServiceType', query),
+      this.param2Query('topicCategory', query),
+      this.param2Query('Subject', query),
+      this.param2Query('Denominator', query),
+      this.param2Query('OrganisationName', query),
+      this.param2Query('keywords', query),
+    ]
+      .filter((n) => {
+        return n != '';
+      })
+      .join(' AND ');
+    let url =
+      dataset.url +
+      '?' +
+      this.hsUtilsService.paramsToURL({
+        request: 'GetRecords',
+        format: 'application/json',
+        language: dataset.language,
+        query: sql,
+        sortby:
+          query.sortby !== undefined && query.sortby != ''
+            ? query.sortby
+            : 'title',
+        limit: dataset.paging.itemsPerPage,
+        start: dataset.datasourcePaging.start,
+      });
+    url = this.hsUtilsService.proxify(url);
+    dataset.datasourcePaging.loaded = false;
+    /*if (dataset.canceler !== undefined) {
+      dataset.canceler.resolve();
+      delete dataset.canceler;
+    }*/
+    //dataset.canceler = $q.defer();
+    this.http
+      .get(url, {
+        //TODO: dataset must be passed to datasetsReceived
+        //timeout: dataset.canceler.promise,
+        //dataset,
+        //extentFeatureCreated,
+        responseType: 'json',
+      })
+      .toPromise()
+      .then(this.datasetsReceived, (e) => {
+        dataset.datasourcePaging.loaded = true;
+      });
+  }
+
+  /**
+   * @private
+   * @function datasetsReceived
+   * @memberof HsMickaBrowserService
+   * @param {Object} j HTTP response containing all the layers
+   * (PRIVATE) Callback for catalogue http query
+   */
+  datasetsReceived(j): void {
+    if (!j.config) {
```

Comment:
In angularjs we could pass extra objects such as dataset and extentFeatureCreated to the request object which got stored in 'config' of the response. It's not working anymore. Saw the TODO in queryCatalog function, but still leaving the comment.